### PR TITLE
Validation for required fields

### PR DESCRIPTION
**Pull request by shushu**
Add isNotEmpty validator to all required fields.

In addition, return FALSE from `getErrors()` if no errors.

### CODE REVIEW

File: includes/EntityValidateBase.php [hunk 1]
```diff
@@ -230,6 +229,9 @@ public function clearErrors() {
    * @return boolean
    */
   public function isNotEmpty($field_name, $value) {
+    if (is_array($value) && isset($value['value'])) {
+      $value = $value['value'];
```
Comment (RoySegall) [inline includes/EntityValidateBase.php:233]: Let's remove this one. The wrapper will handle this for us.

**Comment by amitaibu** [comment]
@RoySegall is this still needed?

**Comment by amitaibu** [comment]
If needed, please re-roll, otherwise close.

**Comment by RoySegall** [comment]
I think i handled this in another PR. Closing.

**Pull request closed without merging**